### PR TITLE
Fix "Open replay in new tab" link in replays

### PR DESCRIPTION
**Pull request by MacChaeger**
It wasn't actually opening in a new tab, go figure!

### CODE REVIEW

**Comment by Zarel** [comment]
lel, where do you have links like that? Just detect `target` imo.

**Comment by Zarel** [comment]
Well, I'd prefer some sort of detection that wasn't by text. `data-target` attribute works.

**Comment by MacChaeger** [comment]
How do I access that, though? Variable names can't have hyphens in them.

**Comment by Zarel** [comment]
attribute, not variable.

The link would look like `<a href="..." data-target="new-window">...</a>`

And the check would look like `this.dataset.target === 'new-window'`

**Comment by MacChaeger** [comment]
Yeah, I meant that `this.data-target` didn't work, and what you've suggested doesn't work. Neither does checking `this.data('target')`

**Comment by Zarel** [comment]
`this.dataset.target` definitely does work...

**Comment by Zarel** [comment]
Can you just tell me where you have links like that?

**Comment by MacChaeger** [comment]
https://github.com/Zarel/Pokemon-Showdown-Client/blob/master/js/client.js#L836

**Comment by MacChaeger** [comment]
I modified the link to be `<a href=' + replayLink + ' target="_blank" data-target="new-window">Open replay in new tab</a>` and the check to be `!target || target === 'search' || this.dataset.target === 'new-window'` and it still redirects to joining the room instead of opening in a new tab, which is the behavior I'm trying to fix.

**Comment by Zarel** [comment]
Oh, that one's just because `data-target` isn't a whitelisted attribute.

**Comment by Zarel** [comment]
I'd say set it as a class. Name it `no-panel-intercept` or something.

**Comment by MacChaeger** [comment]
OK that works. Lesson learned.

File: js/client.js [hunk 1]
```diff
@@ -1235,7 +1235,7 @@
 						if (target === 'appeal' || target === 'appeals') target = 'view-help-request--appeal';
 						if (target === 'report') target = 'view-help-request--report';
 						if (this.host === 'replay.pokemonshowdown.com') {
-							if (!target || target === 'search') {
+							if (!target || target === 'search' || this.className === 'no-panel-intercept') {
```
Comment (Zarel) [inline js/client.js]: I would put the check outside of the `this.host === 'replay.pokemonshowdown.com'` check.

**Comment by Zarel** [comment]
This pull request **introduces 1 alert** when merging f6be54aa23234e84fa78de28880afca2aef0b017 into 810da3b7d6b3adc81be0fff5f293caad24626434 - [view on LGTM.com](https://lgtm.com/projects/g/Zarel/Pokemon-Showdown-Client/rev/pr-d251094dd1a91d6ca6f4ba6963bac6c49196f048)

**new alerts:**

* 1 for Comparison between inconvertible types

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: js/client.js [hunk 2]
```diff
@@ -1228,7 +1228,7 @@
 				if (this.className === 'closebutton') return; // handled elsewhere
 				if (this.className.indexOf('minilogo') >= 0) return; // handled elsewhere
 				if (!this.href) return; // should never happen
-				if (this.host === 'play.pokemonshowdown.com' || this.host === 'replay.pokemonshowdown.com' || this.host === 'psim.us' || this.host === location.host) {
+				if ((this.host === 'play.pokemonshowdown.com' || this.host === 'replay.pokemonshowdown.com' || this.host === 'psim.us' || this.host === location.host) && this.className !== 'no-panel-intercept') {
```
Comment (Zarel) [inline js/client.js]:
```suggestion
				if ([
					'play.pokemonshowdown.com', 'replay.pokemonshowdown.com', 'psim.us', location.host,
				].includes(this.host) && this.className !== 'no-panel-intercept') {
```

Line length is a bit long; edit for readability.

**Comment by Zarel** [comment]
I'm going to have to do this myself to avoid a merge conflict. :|

**Pull request merged**